pkgs/by-name/sc/scalingo/package.nix

```
@@ -21,8 +22,12 @@ buildGoModule rec {
     export HOME=$TMPDIR
   '';
 
+  nativeBuildInputs = [ installShellFiles ];
   postInstall = ''
     rm $out/bin/dists
+    installShellCompletion --cmd scalingo \
+     --bash cmd/autocomplete/scripts/scalingo_complete.bash \
+     --zsh cmd/autocomplete/scripts/scalingo_complete.zsh
```

```suggestion
  ''
  ++ lib.optionalString (stdenv.buildPlatform.canExecute stdenv.hostPlatform) ''
    installShellCompletion --cmd scalingo \
     --bash cmd/autocomplete/scripts/scalingo_complete.bash \
     --zsh cmd/autocomplete/scripts/scalingo_complete.zsh
```
checking for executability is useful for cross-compilation: https://nixos.org/manual/nixpkgs/unstable/#installshellfiles-installshellcompletion-exampleusage-guarded. without this safeguard, build will error, with it -- completion installation will just be skipped silently

---

I don't get it, isn't it already generated?

---

What's happening here? Are you all hosting your accounts with LLM?

---

defo not me, must've been my sleep deprivation. apologies for causing confusion :pray: OP, feel free to undo this change, i overthought it

---

I removed the safeguard. 
@Aleksanaa what do you mean with "isn't it already generated"? is the safeguard automatically added?

---

no, the safeguard is only necessary when you want to generate completions from the compiled program, but in this case they are already provided in the source